clients/admin-ui/src/features/datamap/reporting/DatamapReportTable.tsx

```diff
@@ -273,6 +290,55 @@ export const DatamapReportTable = () => {
     setTotalPages(totalPages);
   }, [totalPages, setTotalPages]);
 
+  // Get custom fields
+  useGetAllCustomFieldDefinitionsQuery();
+  const customFields = useAppSelector(selectAllCustomFieldDefinitions);
+
+  const customFieldColumns = useMemo(() => {
+    // Determine custom field keys by
+    // 1. If they aren't in our expected, static, columns
+    // 2. If they start with one of the custom field prefixes
+    const datamapKeys = datamapReport
+      ? Object.keys(datamapReport.items[0])
+      : [];
+    const defaultKeys = Object.values(COLUMN_IDS);
+    const customFieldKeys = datamapKeys
+      .filter((k) => !defaultKeys.includes(k as COLUMN_IDS))
+      .filter(
+        (k) =>
+          k.startsWith(CUSTOM_FIELD_DATA_USE_PREFIX) ||
+          k.startsWith(CUSTOM_FIELD_SYSTEM_PREFIX)
```

Comment:
this and L317 is a little brittle, but from what I could tell this was the only way to link the datamap custom fields to the actual custom fields